### PR TITLE
Added batch support for IP URL Domain - virusTotal

### PR DESCRIPTION
**Pull request by moishce**
## Status
Ready

## Related Issues
related: demisto/etc#19784

## Description
Added batch support to IP check commands

## Related PRs
List related PRs against other branches:

branch | PR
 ------ | ------
ipinfo-batch | https://github.com/demisto/content/pull/4699
urlscan.io | https://github.com/demisto/content/pull/4703
circl.lu (CVE Search) | https://github.com/demisto/content/pull/4718
Have I been Pwned | https://github.com/demisto/content/pull/4877
AbuseIPDB | https://github.com/demisto/content/pull/4894
IBM X-Force Exchange | https://github.com/demisto/content/pull/4929

## Required version of Demisto
Any

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Code Review

## Dependencies
None

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/virusTotal-batch-support/Integrations/integration-VirusTotal.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/virusTotal-batch-support/Integrations/integration-VirusTotal_CHANGELOG.md)

### CODE REVIEW

**Comment by Itay4** [comment]
@richardbluestone please review

File: Integrations/integration-VirusTotal.yml [hunk 1]
```diff
@@ -353,13 +354,7 @@ script:
...
             md += '\n';
         }
-        return {
-            Type: entryTypes.note,
-            Contents: res.body,
-            ContentsFormat: formats.json,
-            HumanReadable: md,
-            EntryContext: ec
-        };
+       return{Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
```
Comment (Itay4) [inline Integrations/integration-VirusTotal.yml]: please revert to be linted as before

File: Integrations/integration-VirusTotal.yml [hunk 2]
```diff
@@ -379,83 +374,75 @@ script:
...
-        var resolutions = o.resolutions || [];
-        var arrTitle = [{a: detectedUrls, t: 'Detected URL'}, {a: detectedDownloadedSamples, t: 'Detected downloaded sample'},
-            {a: undetectedDownloadedSamples, t: 'Undetected downloaded sample'}, {a: detectedCommunicatingSamples, t: 'Detected communicating sample'},
-            {a: undetectedCommunicatingSamples, t: 'Undetected communicating sample'},{a: detectedReferrerSamples, t: 'Detected referrer sample'},
-            {a: undetectedReferrerSamples, t: 'Undetected referrer sample'}, {a: resolutions, t: 'Resolutions'}];
-        for (var i = 0; i<arrTitle.length; i++) {
-            if (arrTitle[i].a) {
-                md += arrTitle[i].t + ' count: **' + arrTitle[i].a.length + '**\n';
+
+        var ip_list = argToList(ip);
```
Comment (Itay4) [inline Integrations/integration-VirusTotal.yml]:
```suggestion
        var ipList = argToList(ip);
```

File: Integrations/integration-VirusTotal.yml [hunk 3]
```diff
@@ -379,83 +374,75 @@ script:
...
+        var entry_list = [];
```
Comment (Itay4) [inline Integrations/integration-VirusTotal.yml]:
```suggestion
        var entryList = [];
```

**Comment by richardbluestone** [comment]
Files reviewed. I didn't see any changes that were necessary for the YML file.

File: Integrations/integration-VirusTotal.yml [hunk 4]
```diff
@@ -464,146 +459,157 @@ script:
...
+                        var curr = [];
+                        for (var k=0; k<Math.min(arrTitle[j].a.length, sampleSize); k++) {
+                            curr.push(arrTitle[j].a[k]);
+                        }
+                        md += arrToMd(curr) + '\n';
                     }
-                    md += arrToMd(curr) + '\n';
                 }
             }
+            entryList.push({Type: entryTypes.note,Contents: res.body, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec});
```
Comment (Itay4) [inline Integrations/integration-VirusTotal.yml]: please lint as was in the return statement

File: Integrations/integration-VirusTotal.yml [hunk 5]
```diff
@@ -464,146 +459,157 @@ script:
...
+                                ScanID: o.scan_id,
+                                vtLink: o.permalink
+                            },
+                            DetectionEngines: Object.keys(o.scans).length,
+                            PositiveDetections: o.positives
+                        };
+                    }
                 }
             }
+            entryList.push({Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec });
```
Comment (Itay4) [inline Integrations/integration-VirusTotal.yml]: please lint as was in the return statement

File: Integrations/integration-VirusTotal.yml [hunk 6]
```diff
@@ -613,176 +619,178 @@ script:
...
+                    'ReferrerHashes': detected_referrer_samples,
+                    'UnAVDetectedReferrerHashes': undetected_referrer_samples,
+                    'Whois': o.whois,
+                    'Subdomains': domain_siblings,
+                };
+            } else { // not malicious
+                ec[outputPaths.domain] = domain_ec;
+            }
+
+            entryList.push({Type: entryTypes.note, Contents: res.body, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec });
```
Comment (Itay4) [inline Integrations/integration-VirusTotal.yml]: please lint as was in the return statement

**Pull request merged**